tidy3d/plugins/dispersion/fit_web.py

```diff
@@ -0,0 +1,163 @@
+"""Fit PoleResidue Dispersion models to optical NK data based on web service
+"""
+from typing import Tuple, List
+from ...components.types import Literal
+import requests
+from pydantic import BaseModel, PositiveInt, NonNegativeFloat, PositiveFloat, Field
+import numpy as np
+
+from ...components import PoleResidue
+from ...constants import HBAR, MICROMETER
+from ...log import log
+from .fit import DispersionFitter
+
+
+class FitterData(BaseModel):
+    """Data class for request body of Fitter where dipsersion data is input through list"""
+
+    wvl_um: List[float] = Field(
+        ...,
+        title="Wavelengths",
+        description="A list of wavelengths for dispersion data.",
+        units=MICROMETER,
+    )
+    n_data: List[float] = Field(
+        ...,
+        title="Index of refraction",
+        description="Real part of the complex index of refraction.",
+    )
+    k_data: List[float] = Field(
+        ...,
+        title="Extinction coefficient",
+        description="Imaginary part of the complex index of refraction.",
+    )
+
+    num_poles: PositiveInt = Field(
+        1, title="Number of poles", description="Number of poles in model."
+    )
+    num_tries: PositiveInt = Field(
+        50,
+        title="Number of tries",
+        description="Number of optimizations to run with different initial guess.",
+    )
+    tolerance_rms: NonNegativeFloat = Field(
+        0.0,
+        title="RMS error tolerance",
+        description="RMS error below which the fit is successful and result is returned.",
+    )
+    bound_amp: PositiveFloat = Field(
+        100.0,
+        title="Upper bound of oscillator strength",
+        description="Upper bound of oscillator strength in the model.",
+        unis="eV",
+    )
+    bound_f: PositiveFloat = Field(
+        100.0,
+        title="Upper bound of pole frequency",
+        description="Upper bound of pole frequency in the model.",
+        units="eV",
+    )
+    constraint: Literal["hard", "soft"] = Field(
+        "hard",
+        title="Type of constraint for stability",
+        description="Stability constraint enfored on each pole (hard),"
+        " or the summed contribution (soft).",
+    )
+    nlopt_maxeval: PositiveInt = Field(
+        1000,
+        title="Number of inner iterations",
+        description="Number of iterations in each optimization",
+    )
+
+
+class StableDispersionFitter(DispersionFitter):
+
+    """Stable fitter based on web service"""
+
+    def fit(  # pylint:disable=all
+        self,
+        num_poles: PositiveInt = 1,
+        num_tries: PositiveInt = 50,
+        tolerance_rms: NonNegativeFloat = 0.0,
+        bound_amp: PositiveFloat = 100.0 / 2 / np.pi / HBAR,
```

Comment:
Since we covert so much, I think we should define a a static method in `DispersionFitter` for `eV_to_Hz` and `Hz_to_eV` so we can have that conversion in one place when we make major changes to how units are done.